src/server/req_manager.c

```diff
@@ -1561,17 +1612,27 @@ mgr_sched_set(struct batch_request *preq)
 	int	  bad_attr = 0;
 	svrattrl *plist;
 	int	  rc;
+	pbs_sched *psched;
+
+	psched = find_scheduler(preq->rq_ind.rq_manager.rq_objname);
+	if (!psched) {
+		req_reject(PBSE_UNKSCHED, 0, preq);
+		return;
+	}
+	if (validate_sched_attr_vals(preq, psched))
```

Comment:
validate_sched_attr_vals is a validation on batch_request attributes. Which are of type svrattrl. 
Action function do validation on object attributes. I see the action function as propagate/delegate/notify the changes in object to other objects not for input validation. 

If we do validation action function then we need to allocate extra memory.